appshell/appshell_extensions_mac.mm

```diff
@@ -654,46 +654,55 @@ int32 ReadFile(ExtensionString filename, ExtensionString& encoding, std::string&
     if (encoding == "utf8") {
         encoding = "UTF-8";
     }
+    NSString* path = [NSString stringWithUTF8String:filename.c_str()];
+    
+    NSStringEncoding enc;
     int32 error = NO_ERROR;
-
-    try {
-        std::ifstream file(filename.c_str());
-        std::stringstream ss;
-        ss << file.rdbuf();
-        contents = ss.str();
-        std::string detectedCharSet;
+    
+    NSString* fileContents = nil;
+    if (encoding == "UTF-8") {
+        enc = NSUTF8StringEncoding;
+        NSError* NSerror = nil;
+        fileContents = [NSString stringWithContentsOfFile:path encoding:enc error:&NSerror];
+    }
+    
+    if (fileContents)
+    {
+        contents = [fileContents UTF8String];
+        CheckForUTF8BOM(filename, preserveBOM);
```

Comment:
This is the only function call added new in this PR, rest is just revert of my last PR https://github.com/adobe/brackets-shell/pull/610/files#diff-dbb405241ed6688d3fab6d77d1202508